### PR TITLE
Correct Internal Array modules

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2325

File: benchmark/streamly-benchmarks.cabal [hunk 1]
```diff
@@ -651,10 +651,10 @@ benchmark Data.Array
   if flag(limit-build-mem)
     ghc-options: +RTS -M1500M -RTS
 
-benchmark Data.Array.Mut
+benchmark Data.Array.MutArray
   import: bench-options
   type: exitcode-stdio-1.0
-  main-is: Streamly/Benchmark/Data/Array/Mut.hs
+  main-is: Streamly/Benchmark/Data/Array/MutArray.hs
```
Comment (harendra-kumar) [inline benchmark/streamly-benchmarks.cabal:657]: Should be Data.MutArray

File: hie.yaml [hunk 2]
```diff
@@ -20,8 +20,8 @@ cradle:
               component: "bench:Data.Array"
             - path: "./benchmark/Streamly/Benchmark/Data/Array/Generic.hs"
               component: "bench:Data.Array.Generic"
-            - path: "./benchmark/Streamly/Benchmark/Data/Array/Mut.hs"
-              component: "bench:Data.Array.Mut"
+            - path: "./benchmark/Streamly/Benchmark/Data/Array/MutArray.hs"
+              component: "bench:Data.Array.MutArray"
```
Comment (harendra-kumar) [inline hie.yaml:24]: Should be Data.MutArray

File: hie.yaml [hunk 3]
```diff
@@ -98,8 +98,8 @@ cradle:
               component: "test:Data.Stream"
             - path: "./test/Streamly/Test/Data/Array/Generic.hs"
               component: "test:Data.Array.Generic"
-            - path: "./test/Streamly/Test/Data/Array/Mut.hs"
-              component: "test:Data.Array.Mut"
+            - path: "./test/Streamly/Test/Data/Array/MutArray.hs"
+              component: "test:Data.Array.MutArray"
```
Comment (harendra-kumar) [inline hie.yaml:102]: Should be Data.MutArray

File: streamly.cabal [hunk 4]
```diff
@@ -69,7 +69,7 @@ extra-source-files:
     benchmark/Streamly/Benchmark/Data/Array/SmallArray.hs
     benchmark/Streamly/Benchmark/Data/Array.hs
     benchmark/Streamly/Benchmark/Data/Ring/Unboxed.hs
-    benchmark/Streamly/Benchmark/Data/Array/Mut.hs
+    benchmark/Streamly/Benchmark/Data/Array/MutArray.hs
```
Comment (harendra-kumar) [inline streamly.cabal:72]: Should be Data.MutArray

File: streamly.cabal [hunk 5]
```diff
@@ -104,7 +104,7 @@ extra-source-files:
     test/Streamly/Test/Data/Array/Common.hs
     test/Streamly/Test/Data/Array/Generic.hs
     test/Streamly/Test/Data/Array.hs
-    test/Streamly/Test/Data/Array/Mut.hs
+    test/Streamly/Test/Data/Array/MutArray.hs
```
Comment (harendra-kumar) [inline streamly.cabal:107]: Should be Data.MutArray

File: targets/Targets.hs [hunk 6]
```diff
@@ -166,7 +166,7 @@ targets =
             , "pinned_array_cmp"
             ]
       )
-    , ("Data.Array.Mut",
+    , ("Data.Array.MutArray",
```
Comment (harendra-kumar) [inline targets/Targets.hs:169]: Should be Data.MutArray

File: test/Streamly/Test/Data/Array/MutArray.hs [hunk 7]
```diff
@@ -1,4 +1,4 @@
-module Streamly.Test.Data.Array.Mut (main) where
+module Streamly.Test.Data.Array.MutArray (main) where
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Array/MutArray.hs:1]: Should be Data.MutArray

File: test/Streamly/Test/Data/Array/MutArray.hs [hunk 8]
```diff
@@ -23,15 +23,15 @@ import Test.QuickCheck.Monadic (monadicIO, assert)
...
+import qualified Streamly.Internal.Data.MutArray as MArray
 import qualified Streamly.Internal.Data.Stream as Stream
 import qualified Test.Hspec as Hspec
 
 maxTestCount :: Int
 maxTestCount = 100
 
 moduleName :: String
-moduleName = "Data.Array.Mut"
+moduleName = "Data.Array.MutArray"
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Array/MutArray.hs:34]: Should be Data.MutArray

File: test/streamly-tests.cabal [hunk 9]
```diff
@@ -248,11 +248,11 @@ test-suite Data.Stream
...
   ghc-options: -main-is Streamly.Test.Data.Stream.main
 
-test-suite Data.Array.Mut
+test-suite Data.Array.MutArray
   import: test-options
   type: exitcode-stdio-1.0
-  main-is: Streamly/Test/Data/Array/Mut.hs
-  ghc-options: -main-is Streamly.Test.Data.Array.Mut.main
+  main-is: Streamly/Test/Data/Array/MutArray.hs
+  ghc-options: -main-is Streamly.Test.Data.Array.MutArray.main
```
Comment (harendra-kumar) [inline test/streamly-tests.cabal:255]: Should be Data.MutArray

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — APPROVED** [review]

**Review by adithyaov — APPROVED** [review]
LGTM

**Pull request merged**